sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -5268,6 +5268,15 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val OPTIONAL_TRAILING_COMMA_IN_NAMED_EXPRESSION_LISTS =
+    buildConf("spark.sql.optionalTrailingCommaInNamedExpressionLists")
```

Comment:
The config name confuses slightly. I would think if it is `true`, the trailing comma is optional otherwise is mandatory  ;-).  
```suggestion
    buildConf("spark.sql.allowTrailingCommaInNamedExpressionLists")
```

Reply:
named expression list is not a user-facing concept, how about `spark.sql.allowTrailingCommaInSelectList`?